src/rpc_client/amount/mod.rs

```diff
@@ -0,0 +1,148 @@
+use evm_rpc_types::Nat256;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::cmp::Ordering;
+use std::fmt;
+use std::marker::PhantomData;
+
+/// `CheckedAmountOf<Unit>` provides a type-safe way to keep an amount of some `Unit`.
+pub struct AmountOf<Unit>(ethnum::u256, PhantomData<Unit>);
+
+impl<Unit> AmountOf<Unit> {
```

Comment:
Subjective: maybe we could change this to `Amount<Unit = ()>`? That way, we could specify `Amount` in place of `AmountOf<()>`, and (imo) this also makes sense in contexts such as `type GasAmount = Amount<GasUnit>`.

Reply:
Good idea, renamed.